### PR TITLE
YLP-2066 migrate historized settings table

### PR DESCRIPTION
**Pull request by Nas84**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [ ] End-to-end tests successful on CI

### CODE REVIEW

**Comment by Nas84** [comment]
Question: 
- some doubts about the testing, regarding the array return,  what do you think ?

File: packages/dumb/src/components/settings/history/history-table-content.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { type TimePrefs } from 'medical-domain'
+import { type IncomingRow } from '../../../models/historized-parameter.model'
+import { transformToViewModels } from './history-table.core'
+import type moment from 'moment-timezone'
+
+interface HistoryTableContentProps {
+  onSwitchToDaily: (date: moment.Moment | Date | number | null) => void
+  rows: IncomingRow[]
+  timePrefs: TimePrefs
+  length: number
```
Comment (TimCanu) [inline packages/dumb/src/components/settings/history/history-table-content.tsx:40]: **❔ question:** ‏I see that this props ends up being used in `packages/dumb/src/components/settings/history/history-table-spanned-row.tsx`. Its value is always `4` (as this is a component being used only once). Maybe, this value could be defined directly into `history-table-spanned-row.tsx` in a const such as `const NUMBER_OF_COLUMNS = 4`? If you disagree, I would  then advise to at least no get the value from `packages/dumb/src/components/settings/history/history-table.tsx` but directly from the file I am commenting on

File: packages/dumb/src/components/settings/history/history-table-content.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+export const HistoryTableContent: FunctionComponent<HistoryTableContentProps> = (props): JSX.Element => {
+  const { onSwitchToDaily, rows, timePrefs, length } = props
+  const allRows = transformToViewModels(rows, timePrefs)
```
Comment (TimCanu) [inline packages/dumb/src/components/settings/history/history-table-content.tsx:45]: **🤓 nitpick:** ‏
```suggestion
  const historizedParameters = buildHistorizedParameters(rows, timePrefs)
```

File: packages/dumb/src/components/settings/history/history-table-content.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { HistoryTableRow } from './history-table-row'
+import React, { type FunctionComponent } from 'react'
+import { type TimePrefs } from 'medical-domain'
+import { type IncomingRow } from '../../../models/historized-parameter.model'
+import { transformToViewModels } from './history-table.core'
+import type moment from 'moment-timezone'
+
+interface HistoryTableContentProps {
+  onSwitchToDaily: (date: moment.Moment | Date | number | null) => void
+  rows: IncomingRow[]
```
Comment (TimCanu) [inline packages/dumb/src/components/settings/history/history-table-content.tsx:38]: **❔ question:** ‏What does `IncomingRow` represents? I don't understand it based on the name

File: packages/dumb/src/components/settings/history/history-table-parameter-change.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import styles from '../diabeloop.css'
+import React, { type FunctionComponent } from 'react'
+import { ChangeType, type HistorizedParameter } from '../../../models/historized-parameter.model'
+import { useTranslation } from 'react-i18next'
+
+interface HistoryTableParameterChangeProps {
+  parameter: HistorizedParameter
+}
+
+const buildIconCssClass = (change: ChangeType): string => {
```
Comment (TimCanu) [inline packages/dumb/src/components/settings/history/history-table-parameter-change.tsx:37]: **❔ question:** ‏Do you plan on unit testing this?

File: packages/dumb/src/components/settings/history/history-table.test.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+              effectiveDate: '2022-11-02T03:00:00Z',
+              changeType: ChangeType.Deleted
+            }
+          ]
+        }
+      ]
+
+      const arr = transformToViewModels(params, preference)
+      // length of parameter array + row with the date
+      expect(arr).toHaveLength(8)
```
Comment (TimCanu) [inline packages/dumb/src/components/settings/history/history-table.test.ts:100]: I think that checking the length is not enough, the real value is in checking the content. Does it answer your question?

File: packages/dumb/src/components/settings/history/history-table-content.tsx (hunk 3 above)
Comment (Nas84) [reply packages/dumb/src/components/settings/history/history-table-content.tsx:38]: indeed and plus  it is a duplication declaration, doing the migration of the device card in parrallel show me this IncomingRow interface is not relevant, it should be ParametersChange

File: packages/dumb/src/components/settings/history/history-table.test.ts (hunk 5 above)
Comment (Nas84) [reply packages/dumb/src/components/settings/history/history-table.test.ts:100]: oh yeah 👍🏽

File: packages/dumb/src/components/settings/test
Comment (TimCanu) [inline packages/dumb/src/components/settings/test:1]: Is this needed? I am confused

File: packages/dumb/src/components/settings/table-test.test
[the remark this answers is not in the document]
Comment (Nas84) [reply packages/dumb/src/components/settings/table-test.test:1]: oh merde je me suis planté dans mon commit

File: packages/dumb/src/components/settings/test
Comment (Nas84) [reply packages/dumb/src/components/settings/test:1]: pareil desole trop tete en lair

**Review by TimCanu — APPROVED** [review]

**Pull request merged**